common/tools.py

```diff
@@ -2323,6 +2323,163 @@ def unity7(self):
+            self.proxy, self.args = None, None
+        else:
+            self.proxy, self.args = self._prepair()
+        self.activate_sleep = False
+        self.started = False
+
+    def _prepair(self):
+        """
+        Try to connect to the given dbus services. If successful it will
+        return a callable dbus proxy and those arguments.
+        """
+        try:
+            if 'DBUS_SESSION_BUS_ADDRESS' in os.environ:
+                sessionbus = dbus.bus.BusConnection(os.environ['DBUS_SESSION_BUS_ADDRESS'])
+            else:
+                sessionbus = dbus.SessionBus()
+            systembus  = dbus.SystemBus()
+        except:
+            return((None, None))
+        des = list(self.DBUS_SLEEP.keys())
+        des.sort()
+        for de in des:
+            dbus_props = self.DBUS_SLEEP[de]
+            try:
+                if dbus_props['bus'] == 'sessionbus':
+                    bus = sessionbus
+                else:
+                    bus = systembus
+                interface = bus.get_object(dbus_props['service'], dbus_props['objectPath'])
+                proxy = interface.get_dbus_method(dbus_props['method'], dbus_props['interface'])
+                return((proxy, dbus_props['arguments']))
+            except dbus.exceptions.DBusException:
+                continue
+        return((None, None))
+
+    def canSleep(self):
+        """
+        Indicate if a valid dbus service is available to put the system into sleep mode.
+        """
+        return(not self.proxy is None or self.is_root)
+
+    def askBeforeQuit(self):
+        """
+        Indicate if SleepMode is ready to fire and so the application
+        shouldn't be closed.
+        """
+        return(self.activate_sleep and not self.started)
+
+    def sleep(self):
+        """
+        Run 'systemctl suspend' if we are root or
+        call the dbus proxy to start the sleep mode.
+        """
+        if not self.activate_sleep:
+            return(False)
+        if self.is_root:
+            syncfs()
+            self.started = True
+            proc = subprocess.Popen(['systemctl', 'suspend'])
+            proc.communicate()
+            return proc.returncode
+        if self.proxy is None:
+            return(False)
+        else:
+            syncfs()
+            self.started = True
+            return(self.proxy(*self.args))
+
+def isRoot():
+    """
+    Check if the current user is root.
+    """
+    return os.geteuid() == 0
+
+def syncfs():
+    """
+    Sync the filesystem to ensure all pending changes are written to disk.
+    """
+    subprocess.call(['sync'])
+
+def checkCommand(command):
+    """
+    Check if a specific command is available on the system.
+    """
+    return subprocess.call(['which', command], stdout=subprocess.PIPE, stderr=subprocess.PIPE) == 0
+
+def processExists(process):
+    """
+    Check if a specific process is running.
+    """
+    try:
+        subprocess.check_output(['pidof', process])
```

Comment:
`pidof` is not contained in the default installation of Debian. The package `procpcs` need to be installed for that. That would be an extra dependency. Try something more standard, e.g. `ps -A | grep process` (in a `subprocess` modules way of course).